druid-shell/src/platform/windows/window.rs

```diff
@@ -649,31 +649,51 @@ impl WndProc for MyWndProc {
                 Some(0)
             }
             //TODO: WM_SYSCOMMAND
-            WM_MOUSEWHEEL => {
+            WM_MOUSEWHEEL | WM_MOUSEHWHEEL => {
                 // TODO: apply mouse sensitivity based on
                 // SPI_GETWHEELSCROLLLINES setting.
                 if let Ok(mut s) = self.state.try_borrow_mut() {
                     let s = s.as_mut().unwrap();
-                    let delta_y = HIWORD(wparam as u32) as i16 as f64;
+                    let system_delta = HIWORD(wparam as u32) as i16 as f64;
                     let mods = get_mod_state();
-                    let delta = if mods.shift {
-                        Vec2::new(-delta_y, 0.)
-                    } else {
-                        Vec2::new(0., -delta_y)
+                    let wheel_delta = match msg {
+                        WM_MOUSEWHEEL if mods.shift => Vec2::new(-system_delta, 0.),
+                        WM_MOUSEWHEEL => Vec2::new(0., -system_delta),
+                        WM_MOUSEHWHEEL => Vec2::new(system_delta, 0.),
+                        _ => unreachable!(),
                     };
-                    s.handler.wheel(delta, mods);
-                } else {
-                    self.log_dropped_msg(hwnd, msg, wparam, lparam);
-                }
-                Some(0)
-            }
-            WM_MOUSEHWHEEL => {
-                if let Ok(mut s) = self.state.try_borrow_mut() {
-                    let s = s.as_mut().unwrap();
-                    let delta_x = HIWORD(wparam as u32) as i16 as f64;
-                    let delta = Vec2::new(delta_x, 0.0);
-                    let mods = get_mod_state();
-                    s.handler.wheel(delta, mods);
+
+                    let mut p = POINT {
+                        x: LOWORD(lparam as u32) as i16 as i32,
+                        y: HIWORD(lparam as u32) as i16 as i32,
+                    };
+                    unsafe {
+                        if ScreenToClient(hwnd, &mut p) == 0 {
+                            warn!(
+                                "ScreenToClient failed: {}",
+                                Error::Hr(HRESULT_FROM_WIN32(GetLastError()))
+                            );
```

Comment:
I'm not sure that returning bad data is better than ignoring the event. Perhaps we shouldn't create the event when we can't get the correct point.